src/main/java/model/pixeldata/PixelData.java

```diff
@@ -0,0 +1,128 @@
+package model.pixeldata;
+
+import java.awt.Color;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A matrix of "pixels" in the form of colors, representing a "image".
+ *
+ * @see Color
+ */
+public final class PixelData implements IReadOnlyPixelData {
+
+  private final List<List<Color>> pixels;
+
+  /**
+   * Creates a PixelData with the given width (amount of pixels horizontally) and height (amount of
+   * pixels vertically).
+   *
+   * @param width  the amount of pixels in width.
+   * @param height the amount of pixels in height.
+   * @throws IndexOutOfBoundsException if width or height is smaller than or equal to zero.
+   */
+  public PixelData(int width, int height) {
+    if ((width <= 0) || (height <= 0)) {
```

Comment:
This is checked twice, once here and another time in "populateMatrix". Maybe we should also check if the width/height is larger than some magic number (read as MaxWidth/MaxHeight). To limit the user.